news-bundle/src/Controller/NewsFeedController.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\Controller;
+
+use Contao\ContentModel;
+use Contao\Controller;
+use Contao\CoreBundle\Asset\ContaoContext;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Image\ImageFactory;
+use Contao\CoreBundle\InsertTag\InsertTagParser;
+use Contao\CoreBundle\Routing\Page\DynamicRouteInterface;
+use Contao\CoreBundle\Routing\Page\PageRoute;
+use Contao\CoreBundle\ServiceAnnotation\Page;
+use Contao\Environment;
+use Contao\File;
+use Contao\FilesModel;
+use Contao\News;
+use Contao\NewsModel;
+use Contao\PageModel;
+use Contao\StringUtil;
+use Contao\UserModel;
+use FeedIo\Feed;
+use FeedIo\Feed\Item;
+use FeedIo\Feed\Item\Author;
+use FeedIo\Feed\Item\Media;
+use FeedIo\FeedIo;
+use FOS\HttpCache\ResponseTagger;
+use Symfony\Component\Filesystem\Path;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @Page(contentComposition=false)
+ */
+class NewsFeedController implements DynamicRouteInterface
+{
+    public const TYPE = 'news_feed';
+
+    private $urlSuffixes = [
+        'atom' => '.xml',
+        'json' => '.json',
+        'rss' => '.xml',
+    ];
+
+    // TODO: Fix content types, maybe make it configurable in the page settings
+    private $contentTypes = [
+        'atom' => 'application/atom+xml',
+        'json' => 'application/feed+json',
+        'rss' => 'application/rss+xml',
+    ];
+
+    public function __construct(private ContaoFramework $framework, private FeedIo $feedIo, private ResponseTagger $responseTagger, private ContaoContext $contaoContext, private ImageFactory $imageFactory, private RequestStack $requestStack, private InsertTagParser $insertTags, private string $projectDir)
```

Comment:
`$responseTagger` service must be optional since it only exists with a reverse proxy. Also need to add respective checks when calling the `->addTags()` method